helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteWorkflow.java

```diff
@@ -85,6 +86,36 @@ public void testDeleteWorkflow() throws InterruptedException {
     Assert.assertNull(_driver.getWorkflowContext(jobQueueName));
   }
 
+  @Test
+  public void testDeleteWorkflowAndRecreate() throws Exception {
```

Comment:
Does this test fail on master (without the change to requiring fullRefresh after cleanup)? If it's possible, it would be great to have a test that recreates the scenario described in the issue

Reply:
Actually no, in a simplistic scenario, the `ResourceConfigChange` is almost immediately next event in the pipeline, so in local setup and mostly in cluster with low-traffic, this issue is not shown (non-deterministic). This issue is evident when couple of `MessageChange` or `TaskCurrentStateChange` events are queued in between `ResourceConfigChange` event.